docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+
+1. Do I want to use Lua to identify mailboxes and verify that users are are authorized to use mail services? **No. Provisioning is done through LDAP.**
+1. Do I want to use Lua to verify passwords that users authenticate with for IMAP/POP3/SMTP in their mail clients? **Yes. Password authentication is done through Lua.**
+1. If the answer is 'yes' to question 1 or 2: are there other methods that better facilitate my use case instead of custom scripts which rely on me being a developer and not just a user? **No. Only HTTP can be used to authenticate against Nextcloud, which is not supported by Dovecot or DMS.**
+
+While it is possible to extend what Nextcloud supports with [Nextcloud apps](https://apps.nextcloud.com/), there is currently a mismatch between what DMS supports and what Nextcloud applications support. This might change in the future. For now, Lua will be used to bridge the gap between DMS and Nextcloud for authentication only (Dovecot passdb), while LDAP will still be used to identify mailboxes and verify authorization (Dovecot userdb).
+
+## Container variables to adjust
+
+Since Docker Mailserver provides minimal support for Lua with Dovecot, there are no environment variables to configure specifically for Lua. Some environment variables that do exist for other aspects of DMS might need to be configured based on the scenario you want to follow.
+
+In the case of the example scenario, the environment variables must be configured as if users will be authenticated against LDAP (to support identification of mailboxes and verifying authorizations through LDAP). See [LDAP Authentication][docs-authldap] for more information. In addition, the following variables are required:
+
+???+ example
+
+    DMS configuration environment variables to let Postfix apply Lua authentication by proxy through Dovecot for authenticated SMTP.
+
+    ```yaml
+    - ENABLE_SASLAUTHD=1
+    - SASLAUTHD_MECHANISMS=rimap
+    - SASLAUTHD_MECH_OPTIONS=127.0.0.1
+    ```
+
+## Modifying Dovecot's configuration
+
+Add the following volume values to the relevant part of `compose.yaml`:
+
+???+ example
+
+    Override and add Dovecot configuration files and lua scripts.
+
+    ```yaml
+        volumes:
+          - ./docker-data/dms/config/dovecot/auth-ldap.conf.ext:/etc/dovecot/conf.d/auth-ldap.conf.ext:ro
+          - ./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf:/etc/dovecot/conf.d/auth-lua-httpbasic.conf:ro
+          - ./docker-data/dms/config/dovecot/lua/:/etc/dovecot/lua/:ro
+    ```
+
+The first volume line [overrides][docs-dovecotoverrideconfiguration] Dovecot's standard LDAP authentication configuration file. The second line [adds][docs-dovecotaddconfiguration] a new configuration file for Lua authentication. The third line adds a directory which will contain Lua scripts. The files and directory will not be changed from inside the container, which is why they are configured as read-only.
```

Comment:
For a Lua directory I would not use individual binding of files. That would require unnecessary edits of `compose.yaml` anytime a Lua library is added. A separate directory was created for Lua files to avoid conflicts with configuration files.

Also note that Lua is not just configuration, but also program code. I do not know if it would be a good idea to reload Dovecot's configuration anytime a Lua script changes. Since it concerns an unsupported/advanced feature, wouldn't it be better to leave reloading of the configuration to the user?

Since this concerns future changes, I would leave it for now as it is. This is community-contributed documentation. It can lag a bit behind official releases and a bit of possible breakage is expected between releases. Once DMS includes the relevant changes, the documentation can be updated to incorporate these where relevant/necessary.

The next documentation commit has the comments included inside the `configuration.yaml` example code.